R/appendRecords.R

```diff
@@ -0,0 +1,163 @@
+appendRecords = function(metadatadir, desiredtz = "", idloc = 1, maxRecordingInterval = NULL) {
+  
+  # TO DO: Add documentation to vignettes on this new parameter
+  # TO DO: make it impossible to use recordingEndSleepHour when appending files?
+  # TO DO: add statistics about merged files in part 2 or store it as a separate report:
+  # - sample rate
+  # - number of merged files
+  # - relative length
+  # - original filenames
+  
+  # Declare local functions:
+  getInfo = function(fn, idloc, tz) {
+    load(fn)
+    hvars = g.extractheadervars(I)
+    if (exists("Clist")) {
+      ID = NA # If Clist exists then ignore this file as it was previously appended
+    } else {
+      ID = extractID(hvars, idloc, fname = I$filename)
+    }
+    start = as.POSIXct(x = M$metashort$timestamp[1], format = "%Y-%m-%dT%H:%M:%S%z", tz = tz)
+    end = as.POSIXct(x = M$metashort$timestamp[nrow(M$metashort)], format = "%Y-%m-%dT%H:%M:%S%z", tz = tz)
+    info = data.frame(ID = ID, start = start, end = end, filename = fn)
+    return(info)
+  }
+  mergePair = function(M1, M2, overlap, tz) {
+    # overlap in epochs
+    if (overlap < 0) { 
+      # there is a gap
+      fillgaps = function(df, overlap, epochSize, pattern) {
+        if (overlap < 0) {
+          N = abs(overlap) - ifelse(pattern == "long", yes = 0, no = 1)
+        } else {
+          N = overlap - 1
+        }
+        NEWSHORT = df[nrow(df), ]
+        NEWSHORT[1:N, ] = NEWSHORT
+        T0 = as.POSIXct(x = NEWSHORT$timestamp[1], format = "%Y-%m-%dT%H:%M:%S%z", tz = tz) + epochSize
+        NEWSHORT$timestamp = POSIXtime2iso8601(seq(T0, T0 + (N * epochSize) - 1, by =  epochSize), tz)
+        if (pattern != "long") {
+          NEWSHORT[, grep(pattern = pattern, x = colnames(NEWSHORT), invert = TRUE, ignore.case = TRUE)] = 0
+        } else {
+          NEWSHORT$EN = 1
```

Comment:
When appending metalong, this line produces an error with certain data formats. For example, it works well with axivity cwa files, but with actigraph gt3x triggers an error. Not sure why, the metalong data has different column names for EN (uppercase in axivity cwa, lowercase in actigraph cwa). I think standardizing column names in metalong now could be problematic for people who already has the part 1 metadata stored. Instead, we could do this case insensitive with:
```
en_cn = grep("en", colnames(df), ignore.case = TRUE)
NEWSHORT[, en_cn] = 1
```

Reply:
I did not intend the function to be used for handling different device brands. Using different brands adds complexities to interpretation and reporting of results. I will now update the function to match files based on both ID and brand. 

If someone needs this to also work with different brands then that could be a new project, where we will have to carefully investigate how to best report the inconsistency in device brand.

Reply:
I have now create a new issue for the en EN inconsistency https://github.com/wadpac/GGIR/issues/816